shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -72,38 +69,48 @@ impl SimpleRedisCache {
                             .get(&table_name)
                             .ok_or_else(|| anyhow!("{table_name} not a caching table"))?;
 
-                        Message::from_frame(Frame::Redis(build_redis_ast_from_sql(
-                            query,
-                            table_cache_schema,
-                        )?))
+                        messages_redis_request.push(Message::from_frame(Frame::Redis(
+                            build_redis_ast_from_sql(query, table_cache_schema)?,
+                        )));
                     }
                     message => bail!("cannot fetch {message:?} from cache"),
-                };
-                message.invalidate_cache();
+                }
             } else {
                 bail!("Failed to get message namespace");
             }
         }
 
-        let mut messages = self
+        let mut messages_redis_response = self
             .cache_chain
             .process_request(
-                Wrapper::new_with_chain_name(messages, self.cache_chain.name.clone()),
+                Wrapper::new_with_chain_name(messages_redis_request, self.cache_chain.name.clone()),
                 "clientdetailstodo".to_string(),
             )
             .await?;
-        for message in &mut messages {
-            info!("Received reply from redis cache {:?}", message);
-            // TODO: Translate the redis reply into cassandra
-            *message = Message::from_frame(Frame::Cassandra(CassandraFrame {
+
+        // Replace cass_request messages with cassandra responses in place.
+        // We reuse the vec like this to save allocations.
+        let mut messages_redis_response_iter = messages_redis_response.iter_mut();
+        for cass_request in &mut messages_cass_request {
+            let _redis_response = messages_redis_response_iter.next();
```

Comment:
What's this doing?

Reply:
Its demonstrating how to get the redis responses that correspond to this cassandra request.
Each cassandra request can have multiple redis responses due to batches getting split up into multiple redis messages.

There is nothing to actually use the redis responses as even before this PR this was left as a TODO: `// TODO: Translate the redis reply into cassandra`

Reply:
Could you add a comment explaining this, all the iterators are a bit hard to follow.

Reply:
In order to understand what the iterators are doing you need to understand how the earlier vecs are constructed.
So I wrote some comments explaining my mental model of how the whole function works.
Does this help?

Also, thanks to writing out these comments I noticed and fixed a bug earlier in this function.

Reply:
Yep that clears it up, thanks!